tests/templates/test_subroutines/test_hilbert_schmidt.py

```diff
@@ -19,6 +19,36 @@
 import pennylane as qml
 
 
+# pylint: disable=protected-access
+def test_flatten_unflatten():
+    """Test the flatten and unflatten methods."""
+    u_tape = qml.tape.QuantumScript([qml.Hadamard("a"), qml.Identity("b")])
+
+    def v_circuit(params):
+        qml.RZ(params, wires=1)
+
+    v_wires = qml.wires.Wires((0, 1))
+    op = qml.HilbertSchmidt([0.1], v_function=v_circuit, v_wires=v_wires, u_tape=u_tape)
+    data, metadata = op._flatten()
+
+    assert data == (0.1,)
+    assert metadata == (
+        ("v_function", v_circuit),
+        ("v_wires", v_wires),
+        ("u_tape", u_tape),
+    )
+
+    assert hash(metadata)
+
+    new_op = type(op)._unflatten(*op._flatten())
+    assert qml.math.allclose(op.data, new_op.data)
+    assert op.hyperparameters["v_function"] == new_op.hyperparameters["v_function"]
+    assert op.hyperparameters["v_wires"] == new_op.hyperparameters["v_wires"]
+    for op1, op2 in zip(op.hyperparameters["u_tape"], new_op.hyperparameters["u_tape"]):
+        assert qml.equal(op1, op2)
+    assert new_op is not op
```

Comment:
We do not check for `assert qml.equal(op, new_op)`, is that because `qml.equal` does not support `HilbertSchmidt` objects, or just they should not be equal?

Reply:
`qml.equal` does not work with `HilbertSchmidt`.

Initially, I was going to add that change too in this PR, but decided there was already too much going on.

Basically, hyperparameters has `v_tape`, and tapes are only considered equal if they have equal identity.